manta-crypto/src/merkle_tree/leaf_map.rs

```diff
@@ -545,7 +581,7 @@ where
     fn remove(&mut self, index: usize) -> bool {
         match self.last_index {
             Some(current_index) if index == current_index => false,
-            _ => !matches!(self.map.remove(&index), None),
+            _ => self.map.remove(&index).is_none(),
```

Comment:
The old one using `!matches!`, I think this need to be `!self.map.remove(&index).is_none()`

Reply:
Yes, you're right. Good catch!